hyp/base.py

```diff
@@ -51,51 +62,46 @@ def links(self, links, linked):
     def build_root_meta(self, meta):
         return meta
 
-    def build_root_links(self, links):
-        rv = {}
-
+    def build_root_links(self, collector, links):
+        # Use the collector to build the links structure
         for key in links:
-            link = self.LINKS[key]
-            association = "%s.%s" % (self.TYPE, key)
-
-            rv[association] = {'type': link['responder'].TYPE}
-            if 'href' in link:
-                rv[association]['href'] = link['href']
-
-        return rv
-
-    def build_root_linked(self, linked):
-        rv = {}
+            collector.use_link(self, key)
```

Comment:
The way I see it, append is more generic. use_link is to signify the link is being used and should be included. During the collection process, it can be called multiple times. It can appear a little strange in this context, but it's really because the input is a hard-coded list of items used through the document.